packages/styled-components/src/models/ComponentStyle.js

```diff
@@ -29,7 +29,11 @@ export default class ComponentStyle {
     this.staticRulesId = '';
     this.isStatic = process.env.NODE_ENV === 'production' && isStaticRules(rules);
     this.componentId = componentId;
-    this.baseHash = hash(componentId);
+
+    // SC_VERSION gives us isolation between multiple runtimes on the page at once
+    // this is improved further with use of the babel plugin "namespace" feature
+    this.baseHash = hash(SC_VERSION + componentId);
```

Comment:
Do you think it's worth running `hash` on a seed value separately and then continue using `phash` here?

Reply:
Will bench it

Reply:
Minor but noticeable improvement 👍